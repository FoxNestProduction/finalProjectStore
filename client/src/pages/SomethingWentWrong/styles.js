export const root = {
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  height: '100%',
  width: '100%',
  textAlign: 'center',
  p: '20px 30px 30px',
};

export const imgWrapper = {
  height: {
    mobile: '280px',
    lgTablet: '320px',
    desktop: '370px',
  },
  width: {
    mobile: '320px',
    lgTablet: '370px',
    desktop: '420px',
  },
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  overflow: 'hidden',
};

export const mainTitle = {
  fontSize: {
    mobile: '28px',
    lgTablet: '36px',
    desktop: '42px',
  },
  fontWeight: 'fontWeightRegular',
  color: 'text.primary',
  mb: '30px',
};

export const mailLink = {
  color: '#6C5FBC',
  fontWeight: 'fontWeightMedium',
  textDecoration: 'none',
};

export const text = {
  fontSize: {
    mobile: '18px',
    lgTablet: '20px',
    desktop: '22px',
  },
  lineHeight: '2em',
  fontWeight: 'fontWeightRegular',
  letterSpacing: '0.17px',
  color: 'text.primary',
  mb: '40px',
  maxWidth: '605px',
};

export const btnWrapper = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  gap: '10%',
};

export const btn = {
  fontFamily: 'Poppins, sans-serif',
  textTransform: 'capitalize',
  lineHeight: '1.5em',
  p: '10px',
  borderRadius: '16px',
  fontSize: {
    mobile: '18px',
    lgTablet: '20px',
    desktop: '22px',
  },
  minWidth: {
    mobile: '110px',
    lgTablet: '130px',
    desktop: '150px',
  },
};

export const homeBtn = {
  color: 'text.primary',
  border: 1,
  transition: 'background-color 0.3s ease, color 0.3s ease',
  '&:hover': {
    backgroundColor: 'primary.hover',
    color: 'text.primaryLight',
  },
};

export const reloadBtn = {
  color: 'text.primaryLight',
  transition: 'background-color 0.3s ease, box-shadow 0.3s ease, color 0.3s ease',
  '&:hover': {
    backgroundColor: 'primary.hover',
    boxShadow: '0px 4px 6px rgba(0, 0, 0, 0.3)',
  },
  '&:active': {
    boxShadow: '0px -1px 4px rgba(0, 0, 0, 0.5)',
    transform: 'translateY(1px)',
    backgroundColor: 'common.white',
    color: '#1C186C',
    border: '1px solid',
    borderColor: 'primary.main',
  },
};
