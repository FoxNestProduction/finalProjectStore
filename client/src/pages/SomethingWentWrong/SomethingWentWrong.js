import React from 'react';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Link from '@mui/material/Link';
import Typography from '@mui/material/Typography';
import { root, imgWrapper, mainTitle, mailLink, text, btnWrapper, btn, reloadBtn, homeBtn } from './styles';
import { ReactComponent as GirlFalls } from '../../assets/svg/smthWrong.svg';

const SomethingWentWrong = () => {
  const handleReload = () => {
    window.location.reload();
  };

  return (
    <Box component="div" sx={root}>
      <Box sx={imgWrapper}>
        <GirlFalls />
      </Box>

      <Typography sx={mainTitle} variant="h3" component="h1">
        Oops... Something went wrong!
      </Typography>
      <Typography sx={text} variant="h4" component="p">
        Please either refresh the page or return home to try again.
        {' '}
        If the issue continues, please
        {' '}
        <Link href="mailto:eatly.supp@gmail.com" sx={mailLink}>
          get in touch.
        </Link>
      </Typography>
      <Box sx={btnWrapper}>
        <Button
          // type="button"
          variant="outlined"
          component="a"
          href="/"
          sx={{ ...btn, ...homeBtn }}
          disableRipple
        >
          Go Home
        </Button>
        <Button
          type="button"
          variant="contained"
          sx={{ ...btn, ...reloadBtn }}
          onClick={handleReload}
          disableRipple
        >
          Reload
        </Button>
      </Box>
    </Box>
  );
};

export default SomethingWentWrong;
