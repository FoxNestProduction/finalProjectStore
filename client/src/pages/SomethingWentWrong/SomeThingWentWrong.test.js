import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter, useNavigate, useLocation } from 'react-router-dom';
import SomethingWentWrong from './SomethingWentWrong';

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useLocation: jest.fn(),
  useNavigate: jest.fn(),
}));

describe('SomethingWentWrong', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('should render SomethingWentWrong', () => {
    const navigateMock = jest.fn();
    useNavigate.mockReturnValue(navigateMock);

    const { asFragment } = render(
      <MemoryRouter>
        <SomethingWentWrong />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();

    const reloadButton = screen.getByText('Reload');

    expect(reloadButton).toBeInTheDocument();
  });
});
