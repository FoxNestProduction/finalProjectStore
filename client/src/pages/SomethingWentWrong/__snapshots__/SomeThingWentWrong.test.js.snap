// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SomethingWentWrong should render SomethingWentWrong 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-17txlf5"
  >
    <div
      class="MuiBox-root css-117hc89"
    >
      <svg>
        smthWrong.svg
      </svg>
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h3 css-1fzg7sk-MuiTypography-root"
    >
      Oops... Something went wrong!
    </h1>
    <p
      class="MuiTypography-root MuiTypography-h4 css-1o5qxe9-MuiTypography-root"
    >
      Please either refresh the page or return home to try again. If the issue continues, please 
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xoiqdk-MuiTypography-root-MuiLink-root"
        href="mailto:eatly.supp@gmail.com"
      >
        get in touch.
      </a>
    </p>
    <div
      class="MuiBox-root css-e7xvdn"
    >
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-zt68by-MuiButtonBase-root-MuiButton-root"
        href="/"
        tabindex="0"
      >
        Go Home
      </a>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-17qa510-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Reload
      </button>
    </div>
  </div>
</DocumentFragment>
`;
